corehq/apps/hqwebapp/static/hqwebapp/js/daterangepicker.config.js

```diff
@@ -48,6 +47,29 @@
 
         $(this).daterangepicker(config);
 
+        var $el = $(this);
+
+        // UCRs
+        var ucr = "userreports/js/configurable_report";
+        if (typeof COMMCAREHQ_MODULES[ucr] !== 'undefined') {
```

Comment:
This is a bit of a rough way to check whether the `userreports/js/configurable_report` has been defined; also this seems like a roundabout way to get code to execute when that module is defined. The straightforward way to do that might be to put this code into that module, or alternatively, put this code into a function that's exported and then called in that module or something along those lines.

I actually am not totally following the thing that "userreports/js/configurable_report" is a heuristic for. If there's a daterangepicker behavior that is conceptually general but you currently only want to do on ucr pages, I might consider a different route, which is making it conditional on something on a data attribute on `$el`. I'm thinking `data-daterangepicker-show-clear="true"` or something like that. (If there's a better concept behind this "feature" than "show clear", use that in the name instead of course.)

Reply:
Got a bit more context from reading the discussion page—didn't realize this had been discussed already when I first opened the link to this PR. I think though my general sentiment and suggested alternatives here hold though.